globals/matchmaking_globals.go

```diff
@@ -9,4 +11,60 @@ type CommonMatchmakeSession struct {
 	ConnectionIDs []uint32 //players in the room, referenced by their connection IDs. This is used instead of the PID in order to ensure we're talking to the correct client (in case of e.g. multiple logins)
 }
 
-var Sessions = []CommonMatchmakeSession{}
\ No newline at end of file
+var Sessions = []CommonMatchmakeSession{}
+
+func DeleteIndex(s []uint32, index int) []uint32 {
+    return append(s[:index], s[index+1:]...)
+}
+
+func RemoveConnectionIDFromRoom(clientConnectionID uint32, gathering uint32){
+	for index, connectionID := range Sessions[gathering].ConnectionIDs {
+		if connectionID == clientConnectionID {
+			Sessions[gathering].ConnectionIDs = DeleteIndex(Sessions[gathering].ConnectionIDs, index)
+		}
+	}
+	if len(Sessions[gathering].ConnectionIDs) == 0{
+		Sessions[gathering] = CommonMatchmakeSession{} //TODO: is there a better way to do this? right now, the index is used as the gathering id, and I'm not sure if there's a good way to remove without reordering.
+	}
+}
+
+func FindClientSessions(clientConnectionID uint32) []uint32 {
+	//TODO: is there any real instance where a client could be in multiple MatchmakeSessions at once? afaik there isn't, but I handle it regardless just in case (if we're sure there isn't, we could remove the handling and speed the code up)
```

Comment:
The only thing like this that appear on servers are Mario Kart 7 communities, but those use `PersistentGathering`. Then the communities have a list of active MatchmakeSessions, so I'm sure there's no way a client could be in multiple MatchmakeSessions